stl/inc/xstring

```diff
@@ -3980,6 +3980,25 @@ public:
         }
     }
 
+#if _HAS_CXX23
+    template <class _Operation>
+    constexpr void resize_and_overwrite(_CRT_GUARDOVERFLOW const size_type _Newsize, _Operation _Op) {
+        if (_Mypair._Myval2._Myres < _Newsize) {
+            _Reallocate_grow_by(
+                _Newsize - size(), [](_Elem* const _New_ptr, const _Elem* const _Old_ptr, const size_type _Oldsize) {

```

Comment:
This calls `size()` and `data()`, but for debug codegen micro-optimization, `<xstring>` generally refers directly to `_Mypair._Myval2._Mysize` and `_Mypair._Myval2._Myptr()`:
https://github.com/microsoft/STL/blob/178b8406a5647507e759442960f17cd33045cdf5/stl/inc/xstring#L3864-L3868
(I am personally not a huge fan of this, since it interferes with code clarity, but users have brought up debug codegen as a concern, and `string` is one of the most important classes, so avoiding unnecessary function calls is probably worth a bit of ugly code.)